tests/systemd_testsuite/binary_tests.pm

```diff
@@ -19,15 +19,13 @@ sub run {
     my ($self) = @_;
     $self->testsuiteinstall;
 
-    #run binary tests
-    assert_script_run 'cd /var/opt/systemd-tests';
-    assert_script_run './run-tests.sh | tee /tmp/testsuite.log', 600;
-    assert_screen("systemd-testsuite-binary-tests-summary");
+    # run binary tests
+    assert_script_run('cd /var/opt/systemd-tests');
+    validate_script_output('./run-tests.sh | tee /tmp/testsuite.log', sub { m/# FAIL:\s*0/ }, timeout => 600);
```

Comment:
can we not rely on exit code?

Reply:
No, we can't. The pipe hides it.

```bash
$ ls --hola | echo $?
0
ls: unrecognized option '--hola'
Pruebe 'ls --help' para más información.
$ ls --hola 
ls: unrecognized option '--hola'
Pruebe 'ls --help' para más información.
$ echo $?
2
```

Reply:
This is why we run `script_output` with `bash -eo pipefail`, see https://github.com/os-autoinst/os-autoinst/blob/72efc09b70846a988e2bc763bb56858d7b2effa7/distribution.pm#L265

You can also try that in a local shell:

```bash
$ ls --hola | cat; echo $?
ls: unrecognized option '--hola'
Try 'ls --help' for more information.
0
$ set -o pipefail
ls --hola | cat; echo $?
ls: unrecognized option '--hola'
Try 'ls --help' for more information.
2
```

You can't use `ls --hola | echo $?` to output the exit code of the `ls` call in the same pipe call. `$?` would output the exit code from the command before that.

Reply:
The [run-tests.sh](https://build.opensuse.org/package/view_file/devel:openSUSE:QA:Tumbleweed/systemd-v237-testsuite/run-tests.sh?expand=1) returns `$success`. Since and assert_script_run was not failing, If `-o pipefail` is working properly, it was then always returning 0. Maybe because `$xpass` never was 0.

So, I we cannot rely on return code as it is anyway.